superduperdb/components/listener.py

```diff
@@ -188,12 +188,24 @@ def schedule_jobs(
         ]
         return out
 
-    def cleanup(self, database: "Datalayer") -> None:
+    def cleanup(self, db: "Datalayer") -> None:
         """Clean up when the listener is deleted.
 
-        :param database: Data layer instance to process.
+        :param db: Data layer instance to process.
         """
-        # TODO - this doesn't seem to do anything
-        if (cleanup := getattr(self.select, 'model_cleanup', None)) is not None:
-            assert not isinstance(self.model, str)
-            cleanup(database, model=self.model.identifier, key=self.key)
+        if db.databackend.db_type == DBType.SQL:
+            # TODO: Ibis does not support delete table.
```

Comment:
https://ibis-project.org/backends/oracle#ibis.backends.oracle.Backend.drop_table

Reply:
I believe that is implemented for main implementations: `db.databackend.conn.drop_table`.